cpu/cc2538/radio/cc2538_rf_radio_ops.c

```diff
@@ -314,22 +315,44 @@ void cc2538_irq_handler(void)
     RFCORE_SFR_RFIRQF0 = 0;
     RFCORE_SFR_RFIRQF1 = 0;
 
+
+    if ((flags_f0 & SFD) && cc2538_sfd_listen) {
+        if (RFCORE->XREG_FSMSTAT1bits.TX_ACTIVE) {
+            cc2538_rf_dev.cb(&cc2538_rf_dev, IEEE802154_RADIO_INDICATION_TX_START);
+        }
+    }
+
     if (flags_f1 & TXDONE) {
         cc2538_rf_dev.cb(&cc2538_rf_dev, IEEE802154_RADIO_CONFIRM_TX_DONE);
     }
 
+    if ((flags_f0 & SFD) && cc2538_sfd_listen) {
+        if (RFCORE->XREG_FSMSTAT1bits.RX_ACTIVE) {
+            cc2538_rf_dev.cb(&cc2538_rf_dev, IEEE802154_RADIO_INDICATION_RX_START);
+        }
+    }
+
     if (flags_f0 & RXPKTDONE) {
         /* CRC check */
         uint8_t pkt_len = rfcore_peek_rx_fifo(0);
         if (rfcore_peek_rx_fifo(pkt_len) & CC2538_CRC_BIT_MASK) {
-            RFCORE_XREG_RFIRQM0 &= RXPKTDONE;
+            /* Disable RX while the frame has not been processed */
+            RFCORE_XREG_RXMASKCLR = 0xFF;
+            if (RFCORE->XREG_FRMCTRL0bits.AUTOACK) {
```

Comment:
hmmmm do you know if this reads the ACK Req bit? Or is it only the "Auto ACK" feature?

Otherwise frames without the ACK Req will always disable SFD detection

Reply:
Ahh hmm no.. I don't think it does

Reply:
I pushed a fix.